squad/core/migrations/0121_auto_20190711_1829.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.21 on 2019-07-11 18:29
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('core', '0120_buildsummary'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='patchsource',
+            name='password',
```

Comment:
not sure if it makes it any more secure, but maybe you should try sth like this: 
https://djangosnippets.org/snippets/1330/

Reply:
Now password is xor'd with `SECRET_KEY` before being stored into the database. Is not as secure as the link you referenced, but avoids storing plain text.

Reply:
Just a note: It was a great idea to obfuscate the password before storing in the database, otherwise it'd be displayed in plain text in https://qa-reports.linaro.org/api/patchsources/. A good thing to do is also obfucate token as well, if added to a patchsource of type github. Or just prevent these fields from be displayed in the API would do it too.

Reply:
you can make the fields 'write only' in the API. This is a good practice for passwords/tokens.